service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/content/OrderIssuedEmailContentProvider.java

```diff
@@ -55,6 +59,27 @@
             .build();
     }
 
+    public Map<String, Object> buildAllocatedJudgeOrderIssuedNotification(CaseDetails caseDetails) {
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        JudgeAndLegalAdvisor judge = getAllocatedJudge(caseData);
+
+        return ImmutableMap.<String, Object>builder()
+            .put("orderType", getTypeOfOrder(caseData, GENERATED_ORDER))
+            .put("callout", buildCallout(caseData))
+            .put("caseUrl", getCaseUrl(caseDetails.getId()))
+            .put("respondentLastName", getFirstRespondentLastName(caseData.getRespondents1()))
+            .put("judgeTitle", judge.getJudgeOrMagistrateTitle())
+            .put("judgeName", judge.getJudgeName())
+            .build();
+    }
+
+    private JudgeAndLegalAdvisor getAllocatedJudge(CaseData caseData) {
+        Optional<Element<GeneratedOrder>> generatedOrder = caseData.getOrderCollection()
+            .stream().reduce((first, last) -> last);
```

Comment:
Re-use here, definitely room for extraction.

Reply:
done :)